### ISSUE TITLE
abilities/variable: Wildcard namespace name is not applied in allPaths()

### ISSUE THREAD

**Issue by philrenaud** [hashicorp/nomad#14034]
allPaths() looks for an exact namespace match when determining if a user should be able to create, list, read, or destroy at a given path.

However, our ACLs can contain wildcards: so both the following would need to be matched against:

```
namespace "*" {
  policy = "read"
  secure_variables { ...
```

```
namespace "namespaces/project123/*" {
  policy = "read"
  secure_variables {...
```

**Comment by ChaiWithJai** [comment]
This example might need to be re-created:

```
namespace "*" {
  policy = "read"
  secure_variables { ...
```

```
namespace "namespaces/project123/*" {
  policy = "read"
  secure_variables {...
```

These namespaces read like paths.

**Comment by philrenaud** [comment]
@ChaiWithJai `_nearestMatchingPath` is used for testing against wildcards in paths; this is specifically about wildcards in namespaces: https://learn.hashicorp.com/tutorials/nomad/access-control-policies

**Comment by ChaiWithJai** [comment]
Gotcha. So when we take a look at `allPaths` lets note this specific line of code:

```
const matchingNamespace = this.namespace ?? 'default';
```

Every time we compute matching paths, we first do a look-up based on the bound namespace property passed into ability. If no namespace is provided, we set a default namespace to `default`. The same way that we set the default namespace to `default` in the `Variable` model.

So now, we need to implement a similar matching algorithm on namespaces that are being passed in. I worked fairly closely with @tgross on the path matching algorithm, and everything was designed to be small re-usable utility functions so we should be in luck here.

I'd recommend a refactor that should look something like:

```
const allPaths () = > {
  const matchingNamespace = nearestMatchingNamespace(this.namespace);
  ...restOfLogic
}

const nearestMatchingNamespace(namespace) {
   if (!namespace) return '*' // or return `default` -- I'll defer this to Tim

   let policyNamespaces = get(policy, 'rulesJSON.Namespaces') || [];

   let matchingNamespace = this._findMatchingNamespace(
          policyNamespaces,
          namespace
        );
}
```

We have a reason to continue extending the `Abstract` ability to use `_findMatchingNamespace` (this handle glob matching) to handle this glob matching behavior.

This should be a simple patch and we can throw a few integration test cases around this. Overall, it should be a pretty simple patch.
